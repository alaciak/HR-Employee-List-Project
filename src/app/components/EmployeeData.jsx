import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';


class EmployeeData extends React.Component {
  static propTypes = {
    history: PropTypes.object.isRequired
  }

  constructor(props) {
    super(props);
  }


  handleOnClickEdit  = e => {
    this.props.history.push(`/edit/${this.props.id}`);
  }

  handleOnClickRemove = e => {
    e.stopPropagation();
    if(typeof this.props.onRemoveEmployee === 'function') {
      this.props.onRemoveEmployee(this.props.id);
    }
  }

  render() {
    return (
      <tr id={ this.props.id } key={ this.props.id } className='employee-data' onClick={ this.handleOnClickEdit }>
          <td>{ this.props.firstname } { this.props.lastname }</td>
          <td>{ this.props.role }</td>
          <td>{ this.props.experience } months</td>
          <td onClick={ this.handleOnClickRemove }>X</td>
          <td className='employee-data employee-data_description'><div>{ this.props.shortdescript }</div></td>
      </tr>
      );
    }
  }

module.exports = withRouter(EmployeeData);
