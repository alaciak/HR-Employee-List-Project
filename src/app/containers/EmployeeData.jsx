import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeEmployee } from '../actions/employeeListActions';
import { translate } from 'react-i18next';
import AlertContainer from 'react-alert';

export class EmployeeData extends React.Component {
  static propTypes = {
    history: PropTypes.object.isRequired
  }

  constructor(props) {
    super(props);
    this.state = {
      alertMessageDispaly: 'none'
    }
  }

  onClickValidate () {
    if (this.props.userType === 'user' && (this.props.employee.position === 'Manager' || this.props.employee.position === 'Admin')) {
      this.props.handleOnClickErrorAlert();
      return false;
    }
    return true;
  }

  handleOnClickEdit = e => {
    e.stopPropagation();
    if(this.onClickValidate()) {
      this.props.history.push(`/edit/${this.props.employee.id}`);
    }
  }

  handleOnClickRemove = e => {
    e.stopPropagation();
    if(this.onClickValidate()) {
      this.props.removeEmployee(this.props.employee.id);
      this.props.handleOnRemoveSuccessAlert();
    }
  }

  render() {
    const { t } = this.props;
    return (
      <tr id={ this.props.employee.id } key={ this.props.employee.id } className='employee-data' onClick={ this.handleOnClickEdit }>
        <td>{ this.props.employee.firstname } { this.props.employee.lastname }</td>
        <td>{ this.props.employee.role }</td>
        <td>{ this.props.employee.experience } {t('employeeList.experienceData')}</td>
        <td className='employee-data_remove'>
          <div onClick={ this.handleOnClickRemove }>X</div>
          <div className='employee-data employee-data_description'>{ this.props.employee.shortdescript }</div>
        </td>
      </tr>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    userType: state.userTypeSelectionReducer.userType
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    removeEmployee: (employeeID) => {
      dispatch(removeEmployee(employeeID));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(translate('translations')(EmployeeData)));
