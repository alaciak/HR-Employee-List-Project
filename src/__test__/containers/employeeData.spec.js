import React from 'react';
import { EmployeeData } from '../../app/containers/EmployeeData.jsx';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import createRouterContext from 'react-router-test-context';

describe('EmployeeData', () => {

  it('should render if the employee object is passed', () => {

    const testEmployee = {
      id: 1,
      firstname: 'testName',
      lastname: 'testLastname',
      position: 'testPosition',
      role: 'testRole',
      experience: 'testExperience',
      shortdescript: 'testShortdescript',
      longdescript: 'testLongDescript'
    };

    const context = createRouterContext();
    EmployeeData.propTypes = {
      router: React.PropTypes.object
    };

    const wrapper = shallow(<EmployeeData employee={ testEmployee }/>, { context });

    expect(toJson(wrapper)).toMatchSnapshot();

  });

});
