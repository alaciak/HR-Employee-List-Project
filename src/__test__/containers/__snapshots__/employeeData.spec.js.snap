// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeData should render if the employee object is passed 1`] = `
<tr
  className="employee-data"
  id={1}
  key="1"
  onClick={[Function]}
>
  <td>
    testName
     
    testLastname
  </td>
  <td>
    testRole
  </td>
  <td>
    testExperience
     month(s)
  </td>
  <td
    className="employee-data_remove"
  >
    <div
      onClick={[Function]}
    >
      X
    </div>
    <div
      className="employee-data employee-data_description"
    >
      testShortdescript
    </div>
  </td>
</tr>
`;
